source/rules.txt

```diff
@@ -0,0 +1,347 @@
+   :ref:`mongodb-analyzer-analyze-linq` guide.
+
+MALinq2001
+~~~~~~~~~~
+
+The ``MALinq2001`` rule detects unsupported LINQ expressions and
+displays a message that contains the unsupported expression.
+
+.. list-table::
+   :header-rows: 0
+   :stub-columns: 1
+   :widths: 35 65
+
+   * - **Category**
+     - ``MongoDB.Analyzer.LINQ``
+
+   * - **Severity**
+     - ``Warning``
+
+   * - **Notes**
+     - | This rule might apply differently depending on the value of
+       | your ``DefaultLinqVersion`` configuration setting. To learn more,
+       | see the :ref:`Analyzer Configuration Options <mongodb-analyzer-configuration-options>`.
+       |
+       | To view examples of unsupported LINQ expressions, see the
+       | `source code testing files <{+product-source-repo+}/blob/main/tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/NotSupportedLinqExpressions.cs>`__.
+
+The following code triggers the ``MALinq2001`` rule to display a message that
+shows the unsupported LINQ expression:
+
+.. code-block:: csharp
+   
+   var results = queryableColl.Where(b => b.GetHashCode() == 1);
+
+The following image displays the ``MALinq2001`` rule and information
+message:
+
+.. figure:: /includes/images/MALinq2001-rule.png
+   :alt: Screenshot of LINQ expression with MALinq2001 rule message
+
+.. tip::
+
+   To learn more about LINQ expressions, see the
+   :ref:`mongodb-analyzer-analyze-linq` guide.
+
+MALinq2002
+~~~~~~~~~~
+
+The ``MALinq2002`` rule detects if an expression is supported only in
+LINQ3 and displays a message that contains the expression.
+
+.. list-table::
+   :header-rows: 0
+   :stub-columns: 1
+   :widths: 35 65
+
+   * - **Category**
+     - ``MongoDB.Analyzer.LINQ``
+
+   * - **Severity**
+     - ``Warning``
+
+   * - **Notes**
+     - | This rule might apply differently depending on the value of
+       | your ``DefaultLinqVersion`` configuration setting. To learn more,
+       | see the :ref:`Analyzer Configuration Options <mongodb-analyzer-configuration-options>`.
+       |
+       | To view examples of supported LINQ3 expressions, see the
+       | `source code testing files <{+product-source-repo+}/blob/main/tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/NotSupportedLinq2.cs>`__.
+
+The following code triggers the ``MALinq2002`` rule to display a message that
+shows the supported LINQ3 expression:
+
+.. code-block:: csharp
+   
+   var results = queryableColl
+       .Where(b => b.Price + 1 == 21)
+       .Where(b => "Re-release: " + b.Title == "Re-release: Siddhartha");
+
+The following image displays the ``MALinq2002`` rule and information
+message:
+
+.. figure:: /includes/images/MALinq2002-rule.png
+   :alt: Screenshot of LINQ expression with MALinq2002 rule message
+
+.. tip::
+
+   To learn more about LINQ expressions, see the
+   :ref:`mongodb-analyzer-analyze-linq` guide.
+
+MAPoco1001
+~~~~~~~~~~
+
+The ``MAPoco1001`` rule evaluates your code for POCO declarations
```

Comment:
It's worth pointing out that we analyze also some of configured serialization settings, like `BsonIgnore`.
Maybe add something like "Some BSON serialization settings are evaluated as well."